pkg/server/api/bundle/v1/service_test.go

```diff
@@ -1079,6 +1079,242 @@ func TestBatchCreateFederatedBundle(t *testing.T) {
+			},
+		},
+		{
+			name: "Update fails if trust domain is server trust domain",
+			bundlesToUpdate: []*types.Bundle{
+				func() *types.Bundle {
+					b := makeValidBundle(t, federatedTrustDomain)
+					b.TrustDomain = "example.org"
+					return b
+				}(),
+			},
+			expectedResults: []*bundlepb.BatchCreateFederatedBundleResponse_Result{
+				{Status: api.CreateStatus(codes.InvalidArgument, `updating a federated bundle for the server's own trust domain (example.org) s not allowed`)},
+			},
+			expectedLogMsgs: []spiretest.LogEntry{
+				{
+					Level:   logrus.ErrorLevel,
+					Message: `Invalid request: updating a federated bundle for the server's own trust domain is not allowed`,
+					Data: logrus.Fields{
+						telemetry.TrustDomainID: "example.org",
+					},
+				},
+			},
+		},
+		{
+			name:            "Update fails if bundle does not exists",
+			bundlesToUpdate: []*types.Bundle{makeValidBundle(t, federatedTrustDomain)},
+			expectedResults: []*bundlepb.BatchCreateFederatedBundleResponse_Result{
+				{
+					Status: api.CreateStatus(codes.NotFound, "bundle not found"),
+				},
+			},
+			expectedLogMsgs: []spiretest.LogEntry{
+				{
+					Level:   logrus.ErrorLevel,
+					Message: "Bundle not found",
+					Data: logrus.Fields{
+						telemetry.TrustDomainID: "another-example.org",
+						logrus.ErrorKey:         "rpc error: code = NotFound desc = no such bundle",
+					},
+				},
+			},
+		},
+		{
+			name:            "Update datastore query fails",
+			bundlesToUpdate: []*types.Bundle{makeValidBundle(t, federatedTrustDomain)},
+			dsError:         errors.New("datastore error"),
+			expectedResults: []*bundlepb.BatchCreateFederatedBundleResponse_Result{
+				{Status: api.CreateStatus(codes.Internal, `unable to update bundle: datastore error`)},
+			},
+			expectedLogMsgs: []spiretest.LogEntry{
+				{
+					Level:   logrus.ErrorLevel,
+					Message: "Unable to update bundle",
+					Data: logrus.Fields{
+						telemetry.TrustDomainID: "another-example.org",
+						logrus.ErrorKey:         "datastore error",
+					},
+				},
+			},
+		},
+		{
+			name: "Invalid bundle provided",
+			bundlesToUpdate: []*types.Bundle{
+				{
+					TrustDomain: federatedTrustDomain.String(),
+					X509Authorities: []*types.X509Certificate{
+						{
+							Asn1: []byte("malformed"),
+						},
+					},
+				},
+			},
+			expectedResults: []*bundlepb.BatchCreateFederatedBundleResponse_Result{
+				{Status: api.CreateStatus(codes.Internal, fmt.Sprintf("failed to convert bundle: unable to parse X.509 authority: %v", expectedX509Err))},
+			},
+			expectedLogMsgs: []spiretest.LogEntry{
+				{
+					Level:   logrus.ErrorLevel,
+					Message: "Failed to convert bundle",
+					Data: logrus.Fields{
+						telemetry.TrustDomainID: "another-example.org",
+						logrus.ErrorKey:         fmt.Sprintf("unable to parse X.509 authority: %v", expectedX509Err),
+					},
+				},
+			},
+		},
+	} {
+		tt := tt
+		t.Run(tt.name, func(t *testing.T) {
+			test.logHook.Reset()
```

Comment:
nit: instead of clean too much things, maybe you can just crete server on each call?